🐛[BUG] fix: 로그인 시 search  500에러

## 🐞 문제 요약
<!-- 어떤 문제가 발생했는지 간결하고 명확하게 설명해주세요. -->
<!-- 필요하다면 에러 메시지, 스크린샷, 콘솔 로그 등을 함께 첨부해 주세요. -->
배포 서버에서 로그인 + search 호출 시 500에러 발생

## 🔎 원인 분석
<!-- 문제의 원인으로 추정되는 부분이 있다면 작성해주세요. -->
```
Caused by: org.hibernate.query.SemanticException: Could not interpret path expression 'wish.id'
```

## ✅ 해결 방법
<!-- 해결한 방법 또는 시도한 해결 방안을 작성해주세요.  -->
부분 수정이나 임시 대응이라면 해당 내용도 함께 적어주세요.

---
### 📌 추가 정보


### 📚 참고 자료
<!-- 관련 문서, 커밋, 링크 등을 남겨주세요. -->

--- a/src/main/java/com/promesa/promesa/common/config/WebConfig.java
+++ b/src/main/java/com/promesa/promesa/common/config/WebConfig.java
@@ -18,7 +18,7 @@ public class WebConfig implements WebMvcConfigurer {
                         "http://13.209.202.120:8081",
                         "https://promesa.co.kr",
                         "http://localhost:8080")
-                .allowedMethods("GET", "POST", "PUT", "DELETE","OPTIONS")
+                .allowedMethods("GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS")
                 .allowedHeaders("*")
                 .allowCredentials(true);
     }

--- a/src/main/java/com/promesa/promesa/domain/item/query/ItemQueryRepository.java
+++ b/src/main/java/com/promesa/promesa/domain/item/query/ItemQueryRepository.java
@@ -223,7 +223,7 @@
         public List<ItemPreviewResponse> searchByItemName(String keyword, Member member) {
             Expression<Boolean> isWished = isWishedExpression(member);
 
-            return queryFactory
+            JPAQuery<ItemPreviewResponse> query = queryFactory
                     .select(Projections.fields(ItemPreviewResponse.class,
                             item.id.as("itemId"),
                             item.saleStatus.as("saleStatus"),
@@ -240,8 +240,16 @@
                     .where(
                             Expressions.stringTemplate("REPLACE({0}, ' ', '')", item.name)
                                     .containsIgnoreCase(keyword)
-                    )
+                    );
 
-                    .fetch();
+            if (member != null) {
+                query.leftJoin(wish).on(
+                        wish.member.id.eq(member.getId())
+                                .and(wish.targetType.eq(TargetType.ITEM))
+                                .and(wish.targetId.eq(item.id))
+                );
+            }
+
+            return query.fetch();
         }
     }

